src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -785,4 +574,217 @@ public static Image getGameIcon(final Window frame) {
+                  + engineVersionOfGameToJoin.toStringFull("_"),
+              "Correct TripleA Engine Not Found", JOptionPane.WARNING_MESSAGE);
+        }
+        return;
+      }
+      // ask user if we really want to do this?
+      final String messageString = "<html>This TripleA engine is version " + ClientContext.engineVersion().getVersion()
+          + " and you are trying to join a game made with version " + engineVersionOfGameToJoin.toString()
+          + "<br>However, this TripleA can only play with engines that are the exact same version as itself (x_x_x_x)."
+          + "<br><br>TripleA now comes with older engines included with it, and has found the engine used by the host. This is a new feature and is in 'beta' stage."
+          + "<br>It will attempt to run a new instance of TripleA using the older engine jar file, and this instance will join the host's game."
+          + "<br>Your current instance will not be closed. Please report any bugs or issues."
+          + "<br><br>Do you wish to continue?</html>";
+      final int answer = JOptionPane.showConfirmDialog(null, messageString, "Run old jar to join hosted game?",
+          JOptionPane.YES_NO_OPTION);
+      if (answer != JOptionPane.YES_OPTION) {
+        return;
+      }
+    }
+    joinGame(description.getPort(), description.getHostedBy().getAddress().getHostAddress(), newClassPath, messengers);
+  }
+
+  // newClassPath can be null
+  private static void joinGame(final int port, final String hostAddressIP, final String newClassPath,
+      final Messengers messengers) {
+    final List<String> commands = new ArrayList<>();
+    ProcessRunnerUtil.populateBasicJavaArgs(commands, newClassPath);
+    commands.add("-D" + TRIPLEA_CLIENT_PROPERTY + "=true");
+    commands.add("-D" + TRIPLEA_PORT_PROPERTY + "=" + port);
+    commands.add("-D" + TRIPLEA_HOST_PROPERTY + "=" + hostAddressIP);
+    commands.add("-D" + TRIPLEA_NAME_PROPERTY + "=" + messengers.getMessenger().getLocalNode().getName());
+    final String javaClass = "games.strategy.engine.framework.GameRunner";
+    commands.add(javaClass);
+    ProcessRunnerUtil.exec(commands);
+  }
+
+  public static String findOldJar(final Version oldVersionNeeded, final boolean ignoreMicro) throws IOException {
+    if (ClientContext.engineVersion().getVersion().equals(oldVersionNeeded, ignoreMicro)) {
+      return System.getProperty("java.class.path");
+    }
+    // first, see if the default/main triplea can run it
+    if (ClientFileSystemHelper.areWeOldExtraJar()) {
+      final String version = System.getProperty(GameRunner.TRIPLEA_ENGINE_VERSION_BIN);
+      if (version != null && version.length() > 0) {
+        Version defaultVersion = null;
+        try {
+          defaultVersion = new Version(version);
+        } catch (final Exception e) {
+          // nothing, just continue
+        }
+        if (defaultVersion != null) {
```

Comment:
if an exception happens the error should be printed... In this case if default version is null the code is not executed - it should be moved inside the try block